src/v/redpanda/admin_server.cc

```diff
@@ -2942,8 +2942,18 @@ admin_server::self_test_start_handler(std::unique_ptr<ss::httpd::request> req) {
         throw co_await redirect_to_leader(*req, model::controller_ntp);
     }
     auto doc = parse_json_body(*req);
+    std::vector<model::node_id> ids;
     cluster::start_test_request r;
     if (!doc.IsNull()) {
+        if (doc.HasMember("nodes")) {
```

Comment:
Just a thought, perhaps we should verify the validity of the JSON before doing anything else (i.e. checking "nodes", "tests", etc. all have the right types, required fields are set, etc., and only then unmarshal into 'r'. I don't feel storngly about it though

Reply:
Agreed, we can use the `json::validator` for this